### PR TITLE
UXG-1268 adds guided setup general

### PR DESCRIPTION
**Pull request by nicmarson**
http://jira.hq.idx.org/browse/UXG-1268

### CODE REVIEW

File: src/vue/backend/src/views/GuidedSetupGeneral.vue [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        }),
+        goBackStep: function () {
+            // to-do: go back in history
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: '/guided-setup/connect/omnibar' })
+        },
+        async goContinue () {
+            await this.verifyAPIkey()
```
Comment (sgreenl5) [inline src/vue/backend/src/views/GuidedSetupGeneral.vue:62]: Why are you verifying the API key? There is not an api key for this page in the designs.

File: src/vue/backend/src/router/index.js [hunk 2]
```diff
@@ -101,8 +102,8 @@ const routes = [
                             },
                             {
                                 path: 'general',
-                                name: 'General Settings'
-                                // component
+                                name: 'General Settings',
+                                component: GuidedSetupGeneral
```
Comment (sgreenl5) [inline src/vue/backend/src/router/index.js:106]: This route does not lead to the content

**Review by sgreenl5 — CHANGES_REQUESTED** [review]

File: src/vue/backend/src/views/GuidedSetupGeneral.vue (hunk 1 above)
Comment (nicmarson) [reply src/vue/backend/src/views/GuidedSetupGeneral.vue:62]: in the A/C it said:
@continue

- Ping API endpoint: TBD

Comment (sgreenl5) [reply src/vue/backend/src/views/GuidedSetupGeneral.vue]: Correct, ping the corresponding api endpoint. Namely, the one that will send the form change to the db. Currently, you are using an action that is specifically for verifying the `APIKey`, which is not included in this ticket.

File: src/vue/backend/src/views/guided-setup/Api.vue [hunk 3]
```diff
@@ -63,7 +62,7 @@ export default {
             this.$router.go(-1)
         },
         goSkipStep: function () {
-            this.$router.push({ path: '/guided-setup/connect/general' })
+            this.$router.push({ path: '/guided-setup/general' })
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/Api.vue:65]: We will want the connect piece to visually separate the major steps in the set up

File: src/vue/backend/src/router/index.js [hunk 4]
```diff
@@ -89,27 +82,27 @@ const routes = [
...
-                                name: 'Omnibar Settings'
-                                // component
-                            }
-                        ]
+                        path: 'welcome',
+                        name: 'Welcome',
+                        component: () => import('@/views/guided-setup/Welcome')
+                    },
+                    {
+                        path: 'api',
```
Comment (sgreenl5) [inline src/vue/backend/src/router/index.js:93]: `api`, `general`, and `omnibar` are missing the `connect/` prefix that visually links them as part of a step

File: src/vue/backend/src/views/guided-setup/General.vue [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    methods: {
+        ...mapActions({
+            generalSettingsStateChange: 'general/generalSettingsStateChange'
+        }),
+        goBackStep: function () {
+            // to-do: go back in history
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: '/guided-setup/omnibar' })
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/General.vue:57]: Missing A/C point: `@skipStep move to Your Listings step`

File: src/vue/backend/src/views/guided-setup/Omnibar.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<template>
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/Omnibar.vue:1]: Out of scope. This file should be done in http://jira.hq.idx.org/browse/UXG-1269

Remove Test code

**Review by sgreenl5 — CHANGES_REQUESTED** [review]

File: src/vue/backend/src/views/guided-setup/General.vue [hunk 7]
```diff
@@ -0,0 +1,66 @@
+<template>
+    <GuidedSetupContentCard
+        :cardTitle="cardTitle"
+        :steps="guidedSetupSteps"
+        :relatedLinks="links"
+        @back-step="goBackStep"
+        @skip-step="goSkipStep"
+        @continue="goContinue">
+        <template v-slot:controls>
+            <GeneralSettings/>
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/General.vue:10]: `Wrapper name` is not getting set in the state

File: src/vue/backend/src/views/guided-setup/General.vue [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        }),
+        goBackStep: function () {
+            // to-do: go back in history
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: '/guided-setup/omnibar' })
+        },
+        goContinue () {
+            setTimeout(() => {
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/General.vue:60]: Missing mock action implementation

File: src/vue/backend/src/views/guided-setup/Api.vue [hunk 9]
```diff
@@ -72,7 +71,7 @@ export default {
             this.success = true
             this.cardTitle = 'Account Connected!'
             setTimeout(() => {
-                this.$router.push({ path: '/guided-setup/connect/general' })
+                this.$router.push({ path: '/guided-setup/general' })
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/Api.vue:74]: Missing the `connect` you added above

File: src/vue/backend/src/views/guided-setup/General.vue [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        goBackStep: function () {
+            // to-do: go back in history
+            this.$router.go(-1)
+        },
+        goSkipStep: function () {
+            this.$router.push({ path: '/guided-setup/listings' })
+        },
+        async goContinue () {
+            await this.saveGeneralSettings()
+            setTimeout(() => {
```
Comment (sgreenl5) [inline src/vue/backend/src/views/guided-setup/General.vue:62]: There is no need for a timeout on this page because there is not a success state.

**Review by sgreenl5 — CHANGES_REQUESTED** [review]

File: src/vue/backend/src/store/modules/general/actions.js [hunk 11]
```diff
@@ -13,7 +13,29 @@ const verifyAPIkey = () => {
     return new Promise(resolve => setTimeout(resolve, 5000))
 }
 
+const saveGeneralSettings = () => {
+    // To-Do: add actual endpoints
+    // GET /idxbroker/v1/admin/settings/general
```
Comment (sgreenl5) [inline src/vue/backend/src/store/modules/general/actions.js:18]: Do not include the api endpoint ticket content, the To-Do by itself is fine

**Review by sgreenl5 — CHANGES_REQUESTED** [review]
Fails to compile
```
'GuidedSetup' is defined but never used 
'GuidedSetupConnectApi' is defined but never used
```

**Review by sgreenl5 — APPROVED** [review]

**Review by ralls — APPROVED** [review]
👍  looks good!

**Review by ErikIDX — APPROVED** [review]

**Pull request merged**